samples/tutorials/mnist/04_Reusable_and_Pre-build_Components_as_Pipeline.ipynb

```diff
@@ -0,0 +1,799 @@
+    "    def predict(project, model, data, version=None):\n",
+    "      \"\"\"Run predictions on a list of instances.\n",
+    "\n",
+    "      Args:\n",
+    "        project: (str), project where the Cloud ML Engine Model is deployed.\n",
+    "        model: (str), model name.\n",
+    "        data: ([[any]]), list of input instances, where each input instance is a\n",
+    "          list of attributes.\n",
+    "        version: str, version of the model to target.\n",
+    "\n",
+    "      Returns:\n",
+    "        Mapping[str: any]: dictionary of prediction results defined by the model.\n",
+    "      \"\"\"\n",
+    "\n",
+    "      service = googleapiclient.discovery.build('ml', 'v1')\n",
+    "      name = 'projects/{}/models/{}'.format(project, model)\n",
+    "\n",
+    "      if version is not None:\n",
+    "        name += '/versions/{}'.format(version)\n",
+    "\n",
+    "      response = service.projects().predict(\n",
+    "          name=name, body={\n",
+    "              'instances': data\n",
+    "          }).execute()\n",
+    "\n",
+    "      if 'error' in response:\n",
+    "        raise RuntimeError(response['error'])\n",
+    "\n",
+    "      return response['predictions']\n",
+    "\n",
+    "    import tensorflow as tf\n",
+    "    import json\n",
+    "    \n",
+    "    mnist = tf.keras.datasets.mnist\n",
+    "    (x_train, y_train),(x_test, y_test) = mnist.load_data()\n",
+    "    x_train, x_test = x_train / 255.0, x_test / 255.0\n",
+    "\n",
+    "    result = predict(\n",
+    "        project=project_id,\n",
+    "        model=model_name,\n",
+    "        data=x_test[0:2].tolist(),\n",
+    "        version=version)\n",
+    "    print(result)\n",
+    "    \n",
+    "    return json.dumps(result)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # Test the function with already deployed version\n",
+    "# deployment_test(\n",
+    "#     project_id=PROJECT_ID,\n",
+    "#     model_name=\"mnist\",\n",
+    "#     version='ver_bb1ebd2a06ab7f321ad3db6b3b3d83e6' # previous deployed version for testing\n",
+    "# )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "deployment_test_op = comp.func_to_container_op(\n",
+    "    func=deployment_test, \n",
+    "    base_image=\"tensorflow/tensorflow:1.15.0-py3\",\n",
+    "    packages_to_install=[\"google-api-python-client==1.7.8\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create your workflow as a Python function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define your pipeline as a Python function. ` @kfp.dsl.pipeline` is a required decoration including `name` and `description` properties. Then compile the pipeline function. After the compilation is completed, a pipeline file is created."
```

Comment:
Please apply the same feedback as notebook 03

Reply:
This has been changed to align with previous notebooks.